src/app/[locale]/dataset-list.tsx

```diff
@@ -15,23 +33,44 @@ interface Props {
 export default function DatasetList({initialSearchResult, locale}: Props) {
   const [selectedLicenses, setSelectedLicenses] = useState<string[]>([]);
   const [selectedPublishers, setSelectedPublishers] = useState<string[]>([]);
+  const [query, setQuery] = useState('');
+  const [offset, setOffset] = useState(0);
+  const [sortBy, setSortBy] = useState<string>(SortBy.Relevance);
+  const [sortOrder, setSortOrder] = useState<string>(SortOrder.Ascending);
 
   const {data, error} = useQuery({
-    queryKey: ['Datasets', {selectedLicenses, selectedPublishers}],
+    queryKey: [
+      'Datasets',
+      {selectedLicenses, selectedPublishers, query, offset, sortBy, sortOrder},
+    ],
     queryFn: async () =>
       clientSearchDatasets({
+        query,
         licenses: selectedLicenses,
         publishers: selectedPublishers,
+        offset,
+        sortBy,
+        sortOrder,
       }),
     // Keep the previous data to prevent flickering after filtering.
     keepPreviousData: true,
     // Only show initial data if no filters are set.
     initialData:
```

Comment:
I totally agree. This is a workaround because I could not get [hydrate](https://tanstack.com/query/v4/docs/react/guides/ssr#using-hydrate) working. I will look into this later.